lib/registries/addon/components/page-link/component.ts

```diff
@@ -0,0 +1,58 @@
+import { tagName } from '@ember-decorators/component';
+import { computed } from '@ember-decorators/object';
+import Component from '@ember/component';
+import { layout } from 'ember-osf-web/decorators/component';
+import { PageManager } from 'ember-osf-web/packages/registration-schema';
+import { getPageParam } from 'ember-osf-web/utils/page-param';
+import template from './template';
+
+@tagName('')
+@layout(template)
+export default class PageLinkComponenet extends Component {
+    // Required
+    pageManager!: PageManager;
+    pageIndex!: number;
+    currentPage!: number;
+    draftId!: string;
+
+    @computed('pageManager.pageHeadingText', 'pageIndex')
+    get pageIndexWithSlug() {
+        return getPageParam(this.pageIndex, this.pageManager.pageHeadingText);
+    }
+
+    @computed('pageManager.{isVisited,pageIsValid}', 'pageIndex', 'currentPage', 'pageIsActive')
+    get pageIcon() {
+        if (this.pageIsActive) {
+            return 'circle-o';
+        }
+        if (this.pageManager.isVisited) {
+            if (this.pageManager.pageIsValid) {
+                return 'check-circle-o';
+            }
+            return 'exclamation-circle';
+        }
+        return 'circle';
+    }
```

Comment:
This is tied to page state and should be mapped from state the same way I've suggested for CSS classes.